animation_nodes/nodes/rotation/c_utils.pyx

```diff
@@ -100,3 +101,25 @@ def getAxisListOfQuaternionList(QuaternionList quaternions, str axis):
         for i in range(output.length):
             output.data[i] = quaternions.data[i].z
     return output
+
+def randomQuaternionList(int seed, int amount):
+    cdef QuaternionList result = QuaternionList(length = amount)
+    cdef Quaternion *values = <Quaternion*>result.data
+    cdef XoShiRo256Plus rng = XoShiRo256Plus(seed)
+    cdef double length
+    cdef double w, x, y, z
+    cdef Py_ssize_t i
+    for i in range(amount):
+        w = 1.0
+        x = rng.nextFloat()
+        y = rng.nextFloat()
+        z = rng.nextFloat()
+
+        values[i].w = <float>w
+        values[i].x = <float>x
+        values[i].y = <float>y
+        values[i].z = <float>z
```

Comment:
No need to cast to float. Also, no need to assign to w, x, y, and z. Just write directly and remove those variables.

Reply:
thanks! you mean, write like this:`values[i] = Quaternion(w,x,y,z)`?